modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C

```diff
@@ -0,0 +1,72 @@
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PINSFVMomentumFriction.h"
+
+registerMooseObject("NavierStokesApp", PINSFVMomentumFriction);
+
+InputParameters
+PINSFVMomentumFriction::validParams()
+{
+  InputParameters params = FVElementalKernel::validParams();
+  params.addClassDescription("Computes a friction force term on fluid in porous media in the "
+                             "Navier Stokes i-th momentum equation.");
+  MooseEnum momentum_component("x=0 y=1 z=2");
+  params.addRequiredParam<MooseEnum>(
+      "momentum_component",
+      momentum_component,
+      "The component of the momentum equation that this kernel applies to.");
+  params.addParam<MaterialPropertyName>("linear_coef_name", "Name of a linear friction coefficient material property.");
+  params.addParam<MaterialPropertyName>("quadratic_coef_name", "Name of a quadratic friction coefficient material property.");
+  params.addParam<MaterialPropertyName>("Darcy_name", "Name of the Darcy coefficients material property.");
+  params.addParam<MaterialPropertyName>("Forchheimer_name", "Name of the Forchheimer coefficients material property.");
+
+  return params;
+}
+
+PINSFVMomentumFriction::PINSFVMomentumFriction(const InputParameters & params)
+  : FVElementalKernel(params),
+  _component(getParam<MooseEnum>("momentum_component")),
+  _linear_friction_matprop(isParamValid("linear_coef_name")
+                               ? &getADMaterialProperty<Real>("linear_coef_name")
+                               : nullptr),
+  _quadratic_friction_matprop(isParamValid("quadratic_coef_name")
+                                  ? &getADMaterialProperty<Real>("quadratic_coef_name")
+                                  : nullptr),
+  _cL(isParamValid("Darcy_name")
+          ? &getADMaterialProperty<RealVectorValue>("Darcy_name")
+          : nullptr),
+  _cQ(isParamValid("Forchheimer_name")
+          ? &getADMaterialProperty<RealVectorValue>("Forchheimer_name")
+          : nullptr),
+  _use_linear_friction_matprop(isParamValid("linear_coef_name")),
+  _use_quadratic_friction_matprop(isParamValid("quadratic_coef_name")),
+  _use_Darcy_friction_model(isParamValid("Darcy_name")),
+  _use_Forchheimer_friction_model(isParamValid("Forchheimer_name"))
+{
+  if (!_use_linear_friction_matprop && !_use_quadratic_friction_matprop &&
+      !_use_Darcy_friction_model && !_use_Forchheimer_friction_model)
+    mooseError("At least one friction model needs to be specified.");
+}
+
+ADReal
+PINSFVMomentumFriction::computeQpResidual()
+{
+  ADReal friction_term;
+  if (_use_linear_friction_matprop)
+    friction_term += (*_linear_friction_matprop)[_qp] * _u[_qp];
+  if (_use_quadratic_friction_matprop)
+    friction_term += (*_quadratic_friction_matprop)[_qp] * _u[_qp] * std::abs(_u[_qp]);
+  if (_use_Darcy_friction_model)
+    friction_term += (*_cL)[_qp](_component) * _u[_qp];
+  if (_use_Forchheimer_friction_model)
+    friction_term += (*_cQ)[_qp](_component) * _u[_qp];
```

Comment:
If the user is using `superficial_rho_u` etc. as their nonlinear variable, then these residuals won't be quite right. The `VarMaterial` way would be to retrieve a `momentum` material property and index that by component

Reply:
Oopsie I’m thinking from a compressible mindset when your object name is still PINSFV. I think u should still change to momentum though right?

Reply:
Hmm I see that the GenFoam paper uses a formulation proportional to velocity, whereas a lot of Pronghorn publications posit a term proportional to momentum. So I suppose you're probably fine here. Or maybe make an optional material property parameter that if supplied is used in the residual, and if not supplied then `_u` is used?

Reply:
ok let's use momentum to be consistent with the closures in Pronghorn for W.